Sources/Client/Views/Play/WithRenderCoordinator.swift

```diff
@@ -7,24 +7,41 @@ import DeltaRenderer
 struct WithRenderCoordinator<Content: View>: View {
   var client: Client
   var content: (RenderCoordinator) -> Content
+  var cancel: () -> Void
 
-  init(for client: Client, @ViewBuilder content: @escaping (RenderCoordinator) -> Content) {
+  init(for client: Client, @ViewBuilder content: @escaping (RenderCoordinator?) -> Content, cancellationHandler cancel: @escaping () -> Void) {
     self.client = client
     self.content = content
+    self.cancel = cancel
   }
 
   @State var renderCoordinator: RenderCoordinator?
+  @State var renderCoordinatorError: RendererError?
 
   var body: some View {
     VStack {
       if let renderCoordinator = renderCoordinator {
         content(renderCoordinator)
       } else {
-        Text("Creating render coordinator")
+        if let error = renderCoordinatorError {
+          Text(error.localizedDescription)
+          
+          Button("Done", action: cancel)
+            .buttonStyle(SecondaryButtonStyle())
+            .frame(width: 150)
+        } else {
+          Text("Creating render coordinator")
+        }
       }
     }
     .onAppear {
-      renderCoordinator = RenderCoordinator(client)
+      do {
+        try renderCoordinator = RenderCoordinator(client)
```

Comment:
I didn't even know this was valid, the more you know! 😅 generally in Swift you'd do `renderCoordinator = try RenderCoordinator(client)`

Reply:
I actually had it like that originally, but at some point I moved the try to the beginning of the line and now I sincerely don't even remember why

Reply:
moved it in the latest commit